pkg/opendsme/contrib/gnrc_netif_opendsme.cpp

```diff
@@ -0,0 +1,188 @@
+    int res;
+    network_uint16_t addr;
+    le_uint64_t ext_addr;
+    uint8_t *addr_ptr = static_cast<uint8_t *>(ext_addr.u8);
+
+    switch (opt->opt) {
+    case NETOPT_MAX_PDU_SIZE:
+        *((uint16_t *)opt->data) = IEEE802154_FRAME_LEN_MAX;
+        res = sizeof(uint16_t);
+        break;
+    case NETOPT_ADDR_LEN:
+        *((uint16_t *)opt->data) = IEEE802154_SHORT_ADDRESS_LEN;
+        res = sizeof(uint16_t);
+        break;
+    case NETOPT_ADDRESS:
+        assert(opt->data_len >= IEEE802154_SHORT_ADDRESS_LEN);
+        m_dsme.getShortAddress(&addr);
+        memcpy(opt->data, &addr, IEEE802154_SHORT_ADDRESS_LEN);
+        res = IEEE802154_SHORT_ADDRESS_LEN;
+        break;
+    case NETOPT_ADDRESS_LONG: {
+        assert(opt->data_len >= IEEE802154_LONG_ADDRESS_LEN);
+        addr_ptr << m_dsme.getAddress();
+        *((be_uint64_t *)opt->data) = byteorder_ltobll(ext_addr);
+        res = IEEE802154_LONG_ADDRESS_LEN;
+        break;
+    }
+    case NETOPT_LINK:
+        assert(opt->data_len >= sizeof(netopt_enable_t));
+        *((netopt_enable_t *)opt->data) = m_dsme.isAssociated()
+                                          ? NETOPT_ENABLE
+                                          : NETOPT_DISABLE;
+        return sizeof(netopt_enable_t);
+    default:
+        res = -ENOTSUP;
+        break;
+    }
+
+    return res;
+}
+
+static int _set(gnrc_netif_t *netif, const gnrc_netapi_opt_t *opt)
+{
+    int res;
+
+    switch (opt->opt) {
+    case NETOPT_LINK:
+        m_dsme.initialize(_pan_coord);
+        m_dsme.start();
+        break;
+    case NETOPT_PAN_COORD:
+        if (*((bool *)opt->data) == true) {
+            _pan_coord = true;
+        }
+        else {
+            _pan_coord = false;
+        }
+        res = sizeof(netopt_enable_t);
+        break;
+#if IS_ACTIVE(CONFIG_IEEE802154_DSME_STATIC_GTS)
+    case NETOPT_GTS_ALLOC: {
+        ieee802154_dsme_alloc_t *alloc = (ieee802154_dsme_alloc_t *)opt->data;
+        uint16_t _addr = byteorder_ntohs(alloc->addr);
+        m_dsme.allocateGTS(alloc->superframe_id, alloc->slot_id, alloc->channel_id,
+                           alloc->tx ? dsme::Direction::TX : dsme::Direction::RX, _addr);
+        res = sizeof(ieee802154_dsme_alloc_t);
+        break;
+    }
+#endif
+    case NETOPT_PROTO:
+        assert(opt->data_len == sizeof(gnrc_nettype_t));
+        res = sizeof(gnrc_nettype_t);
+        break;
+    case NETOPT_GTS_TX:
+        assert(opt->data_len == sizeof(netopt_enable_t));
+        m_dsme.setGTSTransmission(*((netopt_enable_t *)opt->data) ? true : false);
+        res = sizeof(netopt_enable_t);
+        break;
+    case NETOPT_ACK_REQ:
+        assert(opt->data_len == sizeof(netopt_enable_t));
+        m_dsme.setAckReq(*((netopt_enable_t *)opt->data) ? true : false);
+        res = sizeof(netopt_enable_t);
+        break;
+    case NETOPT_SRC_LEN:
+        res = sizeof(uint16_t);
+        break;
+    default:
+        assert(false);
```

Comment:
Looking at other implementations of the same driver function (eg. gnrc_netif_set_from_netdev as used in gnrc_netif_ieee802154.c), maybe returning -ENOTSUP would be the better choice here. (Otherwise, `ifconfig 4 promisc`  panics the device).